### DISCUSSION TITLE
Wait Until - Entity State is x for xx Minutes

### DISCUSSION THREAD

**Discussion by shrub100** [zachowj/node-red-contrib-home-assistant-websocket#929] [Feature Request]
Just wondering if there is a way to get  the 'Wait Until' node to not only wait until a state reaches a specific status, but to wait until it has been in this status for a set period of time. I can do this in the standard 'Current State' node, but I have to put a retrigger after a certain period of time to continually check the state - which is a bit messy.

If there is a better way of doing this in the existing config let me know. Else I think think this would be a useful little tweak :-)

Thanks for all the hard work by the way... loving node-red with HA!

**Comment by tellok** [comment]
need this feature as well

**Comment by scottmckenzie** [comment]
I use a **wait until** node with an **events: state** node wired to a **join** node. The **wait until** node triggers immediately, the **events: state** node triggers after the entity has been in the required state for the required amount of time. Once that occurs the **join** node will pass the message.

**Reply by tellok** [reply]
In some complex scenarios, this may not work well. For example, in a certain situation, if you are waiting, you might need to cancel.
